### DISCUSSION TITLE
Why does the default method in page.tsx have access to the parameters of child components, which are URL parameters, and why can child components manipulate them?

### DISCUSSION THREAD

**Discussion by slow-sheep** [vercel/next.js#67759] [Help]
### Summary

 I am learning Next.js. Here are the child component

![Dingtalk_20240715133058](https://github.com/user-attachments/assets/aa20cf5b-445e-41e4-accf-e0a0e44efba4)
and parent component codes:

![image](https://github.com/user-attachments/assets/291f6753-3953-4e42-9b30-16d201aea003)

I want to know why the parent component gets the searchParams. Is this a built-in feature of Next.js?

**Comment by icyJoseph** [comment]
Indeed, this a built-in feature of the framework. Page's are passed this information as they are rendered. 

This is possible, because, from the framework's point of view, Page's are the leafs of the router tree, aka, that's as far as the framework has any major control over your components, besides client routing hooks, and some functions to collect data from headers and cookies.
